elyra/pipeline/tests/test_airflow_processor.py

```diff
@@ -296,3 +297,22 @@ def test_pipeline_tree_creation(parsed_ordered_dict, sample_metadata, sample_ima
                     if node['app_data'].get(arg):
                         for file in node['app_data'][arg]:
                             assert file in ordered_dict[key]["pipeline_" + arg]
+
+
+def test_collect_envs(processor):
+    pipelines_test_file = 'elyra/pipeline/tests/resources/archive/test.ipynb'
+
+    test_operation = Operation(id='this-is-a-test-id',
+                               type='execution-node',
+                               classifier='airflow',
+                               name='test',
+                               filename=pipelines_test_file,
+                               runtime_image='tensorflow/tensorflow:latest')
+
+    envs = processor._collect_envs(test_operation, cos_secret=None, cos_username='Alice', cos_password='secret')
```

Comment:
This test should also cover the `cos_secret` code path, which would not return `envs['AWS_ACCESS_KEY_ID']` and `envs['AWS_SECRET_ACCESS_KEY']` for null/empty.